src/cls-ah.ts

```diff
@@ -16,72 +16,89 @@
 
 'use strict';
 
-const asyncHook = require('async_hooks');
+import * as asyncHook from 'async_hooks';
+import { Context, Func, Namespace as CLSNamespace } from 'continuation-local-storage';
 
 const wrappedSymbol = Symbol('context_wrapped');
 let contexts = {};
-let current = {};
+let current: Context = {};
 
 asyncHook.createHook({init, before, destroy}).enable();
 
-function Namespace() {}
+const EVENT_EMITTER_METHODS =
+  [ 'addListener', 'on', 'once', 'prependListener', 'prependOncelistener' ];
 
-Namespace.prototype.get = function(k) {
-  return current[k];
-};
+class Namespace implements CLSNamespace {
```

Comment:
Maybe we should consider renaming this `AHNamespace` in the future. `Namespace` was only used to remain in sync with cls. The class hierarchy captures this information now.

Reply:
Renamed to `AsyncHooksNamespace`.